### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by oliviasculley**
- Improves CI Github action
- Changes deploys to happen on `develop` branch
  - This project is doesn't need to have a separate stable branch

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+on:
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (ThorntonMatthewD) [inline .github/workflows/ci.yml]: Would you be open to using v4 instead?

**Review by ThorntonMatthewD — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ThorntonMatthewD) [reply .github/workflows/ci.yml]: I only ask because I saw the deprecation warning under the Checks tab.

Comment (oliviasculley) [reply .github/workflows/ci.yml]: No


Just kidding, thank you for letting me know! :smile:

**Review by ThorntonMatthewD — APPROVED** [review]
:champagne: LGTM!

**Pull request merged**